src/psyclone/psyGen.py

```diff
@@ -4590,81 +4642,142 @@ def _number_handler(self, node, parent):
 class Symbol(object):
     '''
     Symbol item for the Symbol Table. It contains information about: the name
-    of the symbol, its datatype, the number of dimensions and the symbol kind
-    (e.g. local, external, read_arg, write_arg, rw_arg).
-
-    :param name: Name of the symbol.
-    :type name: string
-    :param datatype: Data type of the symbol.
-    :type datatype: string
-    :param dimensions: Dimensions of the symbol (0 represents an scalar
-                       symbol).
-    :type dimensions: list of integers
-    :param kind: Information about the variable declaration attributes.
-    :type kind: string
+    of the symbol, its datatype, the shape and the symbol access (whether it
+    is local, external, read_arg, write_arg or readwrite_arg).
+
+    :param str name: Name of the symbol.
+    :param str datatype: Data type of the symbol.
+    :param int shape: Shape of the symbol (an empty list represents \
+                      a scalar symbol).
+    :param list access: Information to specify if the declaration represents
+                        a local, external or argument variable, if it is an
+                        argument it also specifies if it is read, write or
+                        readwrite.
     '''
-    def __init__(self, name, datatype=None, dimensions=0, kind=None):
+    def __init__(self, name, datatype=None, shape=[], access=None):
+
+        if datatype not in ('real', 'integer'):
+            raise InternalError("Symbol can only be initialized with 'real' "
+                                "or 'integer' datatypes.")
+
+        if not isinstance(shape, list):
+            raise InternalError("Symbol shape attribute must be a list.")
+
+        if False in [isinstance(x, (type(None), int)) for x in shape]:
+            raise InternalError("Symbol shape list elements can only be "
+                                "'integer' or 'None'.")
+
+        if access not in ('local', 'external', 'read_arg', 'write_arg',
```

Comment:
By your example I believe you meant a class variable, and I used a tuple instead of a list to make it a bit more immutable. Also I extended the docstring with a definition of each access type.